src/methods.jl

```diff
@@ -329,3 +329,49 @@ function combine(frames::Vararg{<:AbstractArray}; method = median)
 end
 
 combine(frames::AbstractVector{<:AbstractArray}; method = median) = combine(frames..., method = method)
+
+
+"""
+    subtract_dark!(frame::AbstractArray, dark_frame::AbstractArray; data_exposure = 1, dark_exposure = 1)
+
+In-place version of [`subtract_dark`](@ref)
+
+# See Also
+[`subtract_dark`](@ref)
+"""
+function subtract_dark!(frame::AbstractArray, dark_frame::AbstractArray; data_exposure = 1, dark_exposure = 1)
+    factor = data_exposure / dark_exposure
+    @. frame -= (dark_frame * factor)
```

Comment:
I don't know whether broadcasting is optimising repetitive calculation or not, so I switched to calculate that multiplying factor one step before and then use it next. I think this will save us some extra divisions. Thoughts?

Reply:
This might be fine, even though I'm a bit worried about type stability.  What happens if `frame` is a `Float32`?

Reply:
>  What happens if frame is a `Float32`?

Ok, it seems to work fine:
```julia
julia> A = rand(Float32, 3, 3);

julia> B = rand(Float32, 3, 3);

julia> subtract_dark!(A, B)
3×3 Array{Float32,2}:
  0.352384  -0.14378    0.698295
 -0.394064   0.184986  -0.0990469
 -0.676293   0.786632   0.0116163
```

Reply:
Tried this,
```
frame = rand(Float32, 5, 5)
dark_frame = rand(Float32, 5, 5)
@inferred(subtract_dark!(frame, dark_frame, data_exposure = 7, dark_exposure = 13))
```
This did not error!

Reply:
These comments are quite unsynced!:laughing: